File: 프로그래머스/2/42888. 오픈채팅방/오픈채팅방.py
def solution(record):
    answer = []
    dic = {}
    
    for r in record:
        if r.startswith("Enter") or r.startswith("Change"):
            _, userid, name = r.split()
            dic[userid] = name
            
    for r in record:
        if r.startswith("Enter"):
            _, userid, _ = r.split()
            answer.append(f"{dic[userid]}님이 들어왔습니다.")
        elif r.startswith("Leave"):
            _, userid = r.split()
            answer.append(f"{dic[userid]}님이 나갔습니다.")
        
    return answer

File: 프로그래머스/2/42888. 오픈채팅방/오픈채팅방.swift
import Foundation

func solution(_ record:[String]) -> [String] {
    var dic:[String: String] = [:]
    var answer: [String] = []
    
    for r in record {
        let command = r.components(separatedBy: " ")
        
        if command[0] == "Enter" || command[0] == "Change" {
            dic[command[1]] = command[2]
        }
    }
    
    for r in record {
        let command = r.components(separatedBy: " ")
        switch command[0] {
            case "Enter":
                answer.append("\(dic[command[1]]!)님이 들어왔습니다.")
            case "Leave":
                answer.append("\(dic[command[1]]!)님이 나갔습니다.")
            default:
            break
        }
    }
    return answer
}